File: SDKs/googleapis.com/recommendationengine/v1beta1/python/src/sdk/models/shared/googlecloudrecommendationenginev1beta1gcssource.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class GoogleCloudRecommendationengineV1beta1GcsSource:
    r"""GoogleCloudRecommendationengineV1beta1GcsSource
    Google Cloud Storage location for input content. format.
    """
    
    input_uris: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('inputUris') }})
    json_schema: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('jsonSchema') }})
    

File: SDKs/googleapis.com/recommendationengine/v1beta1/java/lib/src/main/java/openapisdk/models/shared/GoogleCloudRecommendationengineV1beta1GcsSource.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GoogleCloudRecommendationengineV1beta1GcsSource
 * Google Cloud Storage location for input content. format.
**/
public class GoogleCloudRecommendationengineV1beta1GcsSource {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("inputUris")
    public String[] inputUris;
    public GoogleCloudRecommendationengineV1beta1GcsSource withInputUris(String[] inputUris) {
        this.inputUris = inputUris;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("jsonSchema")
    public String jsonSchema;
    public GoogleCloudRecommendationengineV1beta1GcsSource withJsonSchema(String jsonSchema) {
        this.jsonSchema = jsonSchema;
        return this;
    }
}